cmd/update.go

```diff
@@ -15,87 +17,87 @@ var (
-				Port:      port,
-			},
-		}
-
-		// check if required params are set
-		if host == "" || username == "" || password == "" {
-			l.Log.Fatal("requires host, user, and pass to be set")
+		// check that we have at least one host
+		if len(args) <= 0 {
+			log.Error().Msg("update requires at least one host")
+			os.Exit(1)
 		}
 
 		// get status if flag is set and exit
-		if status {
-			err := magellan.GetUpdateStatus(q)
-			if err != nil {
-				l.Log.Errorf("failed toget update status: %v", err)
+		for _, arg := range args {
+			if showStatus {
+				err := magellan.GetUpdateStatus(&magellan.UpdateParams{
+					FirmwarePath:     firmwareUrl,
+					FirmwareVersion:  firmwareVersion,
+					Component:        component,
+					TransferProtocol: transferProtocol,
+					CollectParams: magellan.CollectParams{
+						URI:      arg,
+						Username: username,
+						Password: password,
+						Timeout:  timeout,
+					},
+				})
+				if err != nil {
+					log.Error().Err(err).Msgf("failed to get update status")
+				}
+				return
 			}
-			return
-		}
 
-		// client, err := magellan.NewClient(l, &q.QueryParams)
-		// if err != nil {
-		// 	l.Log.Errorf("failed tomake client: %v", err)
-		// }
-		// err = magellan.UpdateFirmware(client, l, q)
-		err := magellan.UpdateFirmwareRemote(q)
-		if err != nil {
-			l.Log.Errorf("failed toupdate firmware: %v", err)
+			// initiate a remote update
+			err := magellan.UpdateFirmwareRemote(&magellan.UpdateParams{
+				FirmwarePath:     firmwareUrl,
+				FirmwareVersion:  firmwareVersion,
+				Component:        component,
+				TransferProtocol: strings.ToUpper(transferProtocol),
+				CollectParams: magellan.CollectParams{
+					URI:      host,
+					Username: username,
+					Password: password,
+					Timeout:  timeout,
+				},
+			})
+			if err != nil {
+				log.Error().Err(err).Msgf("failed to update firmware")
+			}
 		}
 	},
 }
 
 func init() {
-	updateCmd.Flags().StringVar(&host, "bmc-host", "", "set the BMC host")
-	updateCmd.Flags().IntVar(&port, "bmc-port", 443, "set the BMC port")
-	updateCmd.Flags().StringVar(&username, "user", "", "set the BMC user")
-	updateCmd.Flags().StringVar(&password, "pass", "", "set the BMC password")
+	updateCmd.Flags().StringVar(&username, "username", "", "set the BMC user")
+	updateCmd.Flags().StringVar(&password, "password", "", "set the BMC password")
 	updateCmd.Flags().StringVar(&transferProtocol, "transfer-protocol", "HTTP", "set the transfer protocol")
-	updateCmd.Flags().StringVar(&protocol, "protocol", "https", "set the Redfish protocol")
-	updateCmd.Flags().StringVar(&firmwareUrl, "firmware-url", "", "set the path to the firmware")
-	updateCmd.Flags().StringVar(&firmwareVersion, "firmware-version", "", "set the version of firmware to be installed")
+	updateCmd.Flags().StringVar(&firmwareUrl, "firmware.url", "", "set the path to the firmware")
+	updateCmd.Flags().StringVar(&firmwareVersion, "firmware.version", "", "set the version of firmware to be installed")
```

Comment:
Is there a reason we're changing the dashes to periods?

Reply:
The idea was to make the arg reflect the YAML structure of the config, but maybe that doesn't since?

Reply:
I see.

Personally, I think dashes are sufficient and are more consistent.